packages/perseus/src/widgets/radio/multiple-choice-widget.new.tsx

```diff
@@ -71,280 +71,259 @@ type Props = WidgetProps<
+
+        // Perseus Widget API methods
+        // TODO(LEMS-2994): When we remove the old Radio files, we may need to move some
+        // of the methods from radio.ff.tsx into here, such as getSerializedState, etc.
+        useImperativeHandle(
+            ref,
+            () => ({
+                /**
+                 * Gets the JSON representation of the User Input for AI assistance.
+                 *
+                 * This function is used to provide structured data about the widget's state
+                 * to AI systems that can help users with answering questions.
+                 *
+                 * @returns A structured JSON object representing the widget's prompt
+                 */
+                getPromptJSON: (): RadioPromptJSON => {
+                    return _getPromptJSON(props, props.userInput);
+                },
+            }),
+            [props],
+        );
 
-            return {
-                id: choice.id,
-                content: this.renderContent(content),
-                checked: selected,
-                correct:
-                    choice.correct === undefined
-                        ? !!reviewChoice && !!reviewChoice.correct
-                        : choice.correct,
-                disabled: readOnly,
-                hasRationale: !!choice.rationale,
-                rationale: this.renderContent(choice.rationale),
-                showRationale: rationaleShown,
-                showCorrectness: correctnessShown,
-                isNoneOfTheAbove: !!choice.isNoneOfTheAbove,
-                revealNoneOfTheAbove: !!(questionCompleted && selected),
-                previouslyAnswered,
+        /**
+         * Renders content that may contain nested widgets (currently only passage-refs).
+         * Parses the content, extracts any widgets, and returns a Renderer component
+         * configured with the appropriate context.
+         *
+         * @param content - The content to render (defaults to empty string)
+         * @returns A React element with the rendered content
+         */
+        const renderContent = (content = ""): React.ReactNode => {
+            const {parsedContent, extractedWidgets} =
+                parseNestedWidgets(content);
+
+            // This has been called out as a hack in the past.
+            // We pass in a key here so that we avoid a semi-spurious
+            // react warning when the ChoiceNoneAbove renders a
+            // different renderer in the same place. Note this destroys
+            // state, but since all we're doing is outputting
+            // "None of the above", that is okay. Widgets inside this Renderer
+            // are not discoverable through the parent Renderer's `findWidgets` function.
+            // alwaysUpdate={true} so that passage-refs findWidgets
+            // get called when the outer passage updates the renderer
+            const linterContext: LinterContextProps = {
+                contentType: "radio",
```

Comment:
to preserve radio as a literal type vs as a string
```suggestion
                contentType: "radio" as const,
```